File: stacks/python implementation/stacks.py
class createPerson(object):
	def __init__(self, name, age):
		self.name = name
		self.age = age
		self.next = None

class People(object):
	def __init__(self):
		self.head = None

	def Push(self, name, age):
		Person = createPerson(name, age)

		Person.next = self.head
		self.head = Person
		return self.head

	def Head(self):
		return self.head.name, self.head.age

	def Pop(self):
		self.head = self.head.next
		return self.head

def main():
	people = People()
	people.Push("Christopher", 26)
	people.Push("Tenkorang", 22)
	people.Push("David", 22)
	people.Push("Francis", 26)

	head = people.Pop()
	head = people.Head()
	print(head)		

if __name__ == '__main__':
	main()		

File: stacks/c implentation/stacks.c
#include <stdio.h>
#include <stdlib.h>

struct People{
	char firstLetter;
	int age;
	struct People* next;
};

struct People* createPerson(char firstLetter, int age){
	struct People* person = (struct People*)malloc(sizeof(struct People));

	person->firstLetter = firstLetter;
	person->age = age;
	person->next = NULL;
	return person;
}


struct People* Push(struct People* head,char firstLetter, int age){
	struct People* temp = createPerson(firstLetter,age);

	temp->next = head;
	head = temp;

	return head;
}

void Top(struct People* head){
	printf("%c %d\n",head->firstLetter, head->age);
}

struct People* Pop(struct People* head){
	struct People* temp = head;
	head = temp->next;
	free(temp);
	return head;
}

int main(){
	struct People* head;

	head = Push(head,'T',22);
	head = Push(head,'F',26);
	head = Push(head,'D',22);
	head = Pop(head);
	Top(head);

}